main.py

```diff
@@ -37,14 +37,19 @@ class Config:
 
 class Index(BaseModel):
     text: List[str] = None
+    ids: List[str] = None
     embeddings: torch.Tensor = None
 
     class Config:
         arbitrary_types_allowed = True
 
+class Document(BaseModel):
+    id: str
```

Comment:
`id` is a reserved keyword in python. Probably not an issue because it exists in the namespace of `Document`, but could lead to weird bugs if you overrite the python `id` class by accident.

Reply:
How about `uid` (unique ID)?

Reply:
That works!